libc/src/stdio/printf_core/float_dec_converter.h

```diff
@@ -537,7 +537,7 @@ LIBC_INLINE int convert_float_decimal_typed(Writer *writer,
   }
 
   if (exponent < MANT_WIDTH) {
-    const uint32_t blocks = (precision / BLOCK_SIZE) + 1;
+    const uint32_t blocks = static_cast<uint32_t>(precision / BLOCK_SIZE) + 1;
```

Comment:
you can maybe get more specific here and below; the static_cast can be applied to just BLOCK_SIZE I suspect?